src/bbs2gh/Commands/MigrateRepoCommand.cs

```diff
@@ -2,141 +2,253 @@
+        };
+
+        var bbsProject = new Option<string>("--bbs-project")
+        {
+            IsRequired = false,
+            Description = "The Bitbucket project to migrate."
+        };
+
+        var bbsRepo = new Option<string>("--bbs-repo")
+        {
+            IsRequired = false,
+            Description = "The Bitbucket repository to migrate."
+        };
+
+        var bbsUsername = new Option<string>("--bbs-username")
+        {
+            IsRequired = false,
+            Description = "The Bitbucket username of a user with site admin privileges. If not set will be read from BBS_USERNAME environment variable."
+        };
+
+        var bbsPassword = new Option<string>("--bbs-password")
+        {
+            IsRequired = false,
+            Description = "The Bitbucket password of the user specified by --bbs-username. If not set will be read from BBS_PASSWORD environment variable."
+        };
+
+        // Arguments to import an existing archive
+        var archiveUrl = new Option<string>("--archive-url")
+        {
+            IsRequired = false,
+            Description = "URL used to download Bitbucket Server migration archive. Only needed if you want to manually retrieve the archive from BBS instead of letting this CLI do that for you."
+        };
+        var githubOrg = new Option<string>("--github-org")
+        {
+            IsRequired = false
+        };
+        var githubRepo = new Option<string>("--github-repo")
+        {
+            IsRequired = false
+        };
+        var githubPat = new Option<string>("--github-pat")
+        {
+            IsRequired = false
+        };
+        var wait = new Option("--wait")
+        {
+            Description = "Synchronously waits for the repo migration to finish."
+        };
+        var verbose = new Option("--verbose")
+        {
+            IsRequired = false
+        };
+
+        AddOption(archiveUrl);
+        AddOption(githubOrg);
+        AddOption(githubRepo);
+        AddOption(githubPat);
+
+        AddOption(bbsServerUrl);
+        AddOption(bbsProject);
+        AddOption(bbsRepo);
+        AddOption(bbsUsername);
+        AddOption(bbsPassword);
+
+        AddOption(wait);
+        AddOption(verbose);
+
+        Handler = CommandHandler.Create<MigrateRepoCommandArgs>(Invoke);
+    }
+
+    public async Task Invoke(MigrateRepoCommandArgs args)
+    {
+        if (args is null)
+        {
+            throw new ArgumentNullException(nameof(args));
+        }
+
+        if (!args.BbsServerUrl.HasValue() && !args.ArchiveUrl.HasValue())
+        {
+            throw new ArgumentException("Either --bbs-server-url or --archive-url must be specified.");
+        }
+
+        if (args.BbsServerUrl.HasValue() && args.ArchiveUrl.HasValue())
+        {
+            throw new ArgumentException("Only one of --bbs-server-url or --archive-url can be specified.");
+        }
```

Comment:
Shouldn't these be `OctoshiftCliException`? the difference is that unless they specify `--verbose`, the any exception's message other than `OctoshiftCliException` will be hidden to the user and will only appear in verbose logs but here since we actually want the use to see this error message, we probably want to throw a `OctoshiftCliException`.

Reply:
Ah TIL yes that makes sense